### ISSUE TITLE
[CF Multitenant deploy error] Unable to start application aisaas-srv-${space}, Error acquiring tenant t0

### ISSUE THREAD

**Issue by AlfredWangTJ** [SAP-samples/btp-cap-genai-rag#18]
Hi Experts,

I'm following the instruction to deploy this package for Cloud Foundry Runtime / Multitenant scenario, but I got below error when executing step 3-deploy/2-DeployTheApplication: npm run deploy.

_Application "aisaas-srv-cf_space_us10" staged
Starting application "aisaas-srv-cf_space_us10"...
Error starting application "aisaas-srv-cf_space_us10": Some instances have crashed. Check the logs of your application for more information.
Download the application logs via the dmol command and check them for more information._

![image](https://github.com/SAP-samples/btp-cap-genai-rag/assets/92712829/84379db0-0447-4a07-9eaf-1db3892f52a6)

I downloaded aisaas-srv-cf_space_us10.log for details, This is the error I found but I have no idea what is the cause since I'm new to CAP.

_Error: Error acquiring tenant t0: unknown error_

![image](https://github.com/SAP-samples/btp-cap-genai-rag/assets/92712829/aa5ee2b7-cc49-4ba3-aee9-fdcbf1524060)

The log file is attached:
[aisaas-srv-cf_space_us10.log](https://github.com/SAP-samples/btp-cap-genai-rag/files/15094195/aisaas-srv-cf_space_us10.log)

Could you please suggest how to fix this error, thanks a lot.

**BTW:**
Before this step, another error raised in step 3-deploy/1-BuildTheComponents: npm run build:mbt

_cds.service.protocols.protocol4(...).startsWith is not a function_

I did not capture a screenshot, according to a blog, I updated the version of @sap/cds-dk. This is what I have done.
_# Run in folder: ./multi-tenant/code
npm i @sap/cds-dk_

After executing this npm command, the 'startsWith' error is fixed but I'm not sure whether this could cause the error above when upload and deploy the Application to Cloud Foundry.

**Comment by AlfredWangTJ** [comment]
Hi Kay,

The issue is fixed after build the latest version. Thanks a lot.

I find another issue when creating Subscriptions of the service for 'premium plan'.

_Couldn't subscribe to GenAI Mail Insights cf_space_us10. Please try again. If the problem persists, open a support case for this application._

_Technical Details:
Subscribe failed. Error: Subscribe failed in create callback. Error: CorrelationId: 6a4f8da7-6369-48fc-6777-ca5aed02d380, Please open a ticket for application/service with appName: aisaas-cf_space_us10-BTP###flx. Failed to call callback. URL: PUT https://btp###flx-cf-space-us10-aisaas-srv.cfapps.us10.hana.ondemand.com/-/cds/saas-provisioning/tenant/85b###911. Consumer tenant: 85b###911. Callback response: {"error":{"code":"500","message":"Internal Server Error"}}. status code: 500 INTERNAL_SERVER_ERROR_

![image](https://github.com/SAP-samples/btp-cap-genai-rag/assets/92712829/a7ea7545-e462-43de-85d4-fb64155799c4)
![image](https://github.com/SAP-samples/btp-cap-genai-rag/assets/92712829/fe088674-9bb9-461f-a4dd-292e6e69d41c)

Tried to access the URL: GET https://btp###flx-cf-space-us10-aisaas-srv.cfapps.us10.hana.ondemand.com/-/cds/saas-provisioning/tenant/85b###911 with browser and got '401-Unauthorized' error, this may not help because no credentials is attached when request the URL with browser.

_{"error":{"code":"401","message":"Unauthorized"}}_

This issue only occurred for 'premium plan', the 'default plan' can be subscribed successfully. After subscribe a 'default plan' and assign relevant roles to subaccount user. I got 404 error when try to open the application.

![image](https://github.com/SAP-samples/btp-cap-genai-rag/assets/92712829/edd7ccae-61b7-43b4-a690-e779c9fc2594)

_404 Not Found: Requested route ('genai-email-insights-n75htxcn-aisaas-cf_space_us10.cfapps.us10.hana.ondemand.com') does not exist._

**Comment by kay-schmitteckert** [comment]
Glad that this worked out! Thanks for the feedback.

Which credentials did you use in the credential store? From your personal user? If yes, is 2FA activated? Usually this causes such problems. If not, could you double check your credentials that they are correct?

Best
Kay

**Comment by AlfredWangTJ** [comment]
Hi Kay,

I'm using the I-User email and the global Windows AD password (Because SSO enabled, NO separate password for BTP is set), besides, NO 2FA is enabled.
I have double checked the credential store, the configuration follows the instruction, so I think there maybe some other potential issues during the Subscription process.

Thanks and Regards.

**Comment by kay-schmitteckert** [comment]
Thanks for the confirmation. Did you entitle the service "Cloud Management Service" with "central plan"?

**Comment by AlfredWangTJ** [comment]
Hi Kay,

Thanks a lot for the timely reply. I checked the entitlements and seems I missed out 'central plan' for 'Cloud Management Service' in my subaccount. After adding this entitlement and re-deploy the App, I can subscribe to 'premium plan' successfully.

Another thing I notice that the BTP platform may have some kind of cache related to a session, if I subscribe immediately after the deployment, this error will persist. Only after I clear the cookies and re-login to BTP platform (or open a new In-private window in browser) could I make the subscription successfully.

However, after subscription and add roles to consumer subaccount, I got an HTTP 500 error when authorizing the access, could this be some kind of issue related to credential store?

![image](https://github.com/SAP-samples/btp-cap-genai-rag/assets/92712829/c6527693-7a28-4365-9a5f-28a768b41834)
![image](https://github.com/SAP-samples/btp-cap-genai-rag/assets/92712829/e6316df3-f389-4fa4-b1fb-c40c3f993d02)
![image](https://github.com/SAP-samples/btp-cap-genai-rag/assets/92712829/0f012034-e5f6-4849-a50d-b7ee66cb4d78)

I have downloaded the application log for aisaas-cf_space_us10 and caputre a nerwork .har file for reference.
[aisaas-cf_space_us10-2024-04-26 03_27_42.067+0000.txt](https://github.com/SAP-samples/btp-cap-genai-rag/files/15125053/aisaas-cf_space_us10-2024-04-26.03_27_42.067%2B0000.txt)
[genai-email-insights.har.txt](https://github.com/SAP-samples/btp-cap-genai-rag/files/15125058/genai-email-insights.har.txt)

I did not get much error details from the log this time, only a message 'query does not exist for request url /aisaasui/index-cdn.html' with level 'log' but not 'error'.

![image](https://github.com/SAP-samples/btp-cap-genai-rag/assets/92712829/df39cb9b-b13a-4921-a29e-edcc77fcad72)
